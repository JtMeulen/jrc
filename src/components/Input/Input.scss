@use '../../style/lib.scss' as *;
@use '../../style/mixins.scss';

.input {
  @include mixins.font-family();
}