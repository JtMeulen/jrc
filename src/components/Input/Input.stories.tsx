import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Input from "./Input";

export default {
  title: "Components/Input",
  component: Input,
} as Meta<typeof Input>;

const Template: StoryFn<typeof Input> = (args) => <Input {...args} />;

export const LabelledAndPlaceholder = Template.bind({});
LabelledAndPlaceholder.args = {
  label: "First name",
  placeholder: "Type your name here",
};

export const WithValue = Template.bind({});
WithValue.args = {
  value: 'Joey',
  label: "First name",
  placeholder: "Type your name here",
};

export const NoLabel = Template.bind({});
NoLabel.args = {};
