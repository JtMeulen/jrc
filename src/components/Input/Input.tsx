import React, { useState, useEffect, HTMLInputTypeAttribute } from "react";
import cn from "classnames";
import "./Input.scss";

export interface InputProps {
  name: string;
  type?: HTMLInputTypeAttribute;
  label?: string;
  placeholder?: string;
  value?: string;
  required?: boolean;
  disabled?: boolean;
}

// TODO: add custom validator

const Input = ({
  label,
  placeholder,
  type,
  name,
  value = "",
  required = false,
  disabled = false,
}: InputProps) => {
  const [_value, setValue] = useState(value);

  useEffect(() => {
    setValue(value);
  }, [value]);

  return (
    <>
      {label && (
        <label id={`${"TODO:RANDOM"}-label`} className="label">
          {label}
        </label>
      )}
      <input
        data-testid="jrc-input"
        className={cn("input")}
        placeholder={placeholder}
        type={type}
        name={name}
        labelled-by={label && `${"TODO:RANDOM"}-label`} //TODO: UseRef good for this?
        value={_value}
        onChange={(e) => setValue(e.target.value)}
        required={required}
        disabled={disabled}
      ></input>
    </>
  );
};

export default Input;
