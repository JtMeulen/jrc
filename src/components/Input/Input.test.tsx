import React from "react";
import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Input from "./Input";

describe("Input", () => {
  test("renders without error", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });
});
