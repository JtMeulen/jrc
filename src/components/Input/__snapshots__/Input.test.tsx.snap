// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with label and attributes 1`] = `
<div>
  <label
    class="label"
    id="TODO:RANDOM-label"
  >
    Input label
  </label>
  <input
    class="input"
    data-testid="jrc-input"
    disabled=""
    labelled-by="TODO:RANDOM-label"
    name="Input field"
    placeholder="Placeholder text"
    required=""
    type="text"
    value="I am value"
  />
</div>
`;

exports[`Input renders without error 1`] = `
<div>
  <input
    class="input"
    data-testid="jrc-input"
    name="Input field"
    type="text"
    value=""
  />
</div>
`;
